spec/unit/validator_spec.rb

```diff
@@ -788,6 +788,28 @@ def podspec_path(name = 'JSONKit', version = '1.4')
           @validator.instance_variable_set(:@installer, installer)
           @validator.send(:add_app_project_import)
         end
+
+        it 'should share pod target xcscheme' do
+          @validator.send(:create_app_project)
+          pods_project = Xcodeproj::Project.new(@validator.validation_dir + 'Pods/Pods.xcodeproj')
+          pod_target = fixture_pod_target('coconut-lib/CoconutLib.podspec')
+          installer = stub(:pod_targets => [pod_target])
+          installer.stubs(:pods_project).returns(pods_project)
+          File.stubs(:exist?).returns(true)
```

Comment:
can you pass exactly what you expect here? Why are we blindly mocking this?

Reply:
These tests don't appear to be testing anything in particular. If we mock out this above then of course the test would return `true`. 

I think you need more of an integration test which is the problem with this class and its tests in general.

Reply:
Yeah, the test could be improved.  I probably won't have time in the near future to get this working as an integration test.  If we could mock the specific file name rather than mocking the whole method, would that be "good enough" for now? (I'm not sure if that's possible with these mocking tools, though.)

Reply:
I looked at this a little more today.  It is hard to improve this test since it doesn't actually install the pods.  The test for the initial fix for #6558 is also pretty sorry as it stubs out the method `@validator.stubs(:shares_pod_target_xcscheme?).returns(false)`

Since my fix edits the code in that file, I at least tried to stub the dependency within that method (does the file exist or not) rather than stubbing out the whole method.